### DISCUSSION TITLE
Check code via compiling button

### DISCUSSION THREAD

**Discussion by Foxelyss** [pybricks/support#685] [Ideas]
Hi! I am started work with pybricks and found that there is no button to check code via compilation like in Arduino IDE . Adding such a button is very good enhancement,because everyone will know are their program in fit's in memory or not and there will not be need to get hub running all time for finding some typing errors in code.

**Comment by laurensvalk** [comment]
Try pressing F2:

![image](https://user-images.githubusercontent.com/12326241/179001243-0abe43c4-d681-4dca-bcf5-f7a049715d45.png)

As for program size, the available space depends on which hub you use.

**Reply by Foxelyss** [reply]
Thanks.But how do starters figure it out!?All actions have buttons respectively except for this one.Its like some secret feature.

**Reply by dlech** [reply]
I started #695 as an official feature request. If you have any ideas for this, please add them there.
